### PR TITLE
Docs/key value tags for resources

### PR DESCRIPTION
**Pull request by trujillo-adam**
### What

This PR updates the documentation to describe tags stored as key-value pairs. It describes the UI for creating and managing tags while creating project and workspaces. 

- Implements [IPE-867](https://hashicorp.atlassian.net/browse/IPE-867). 
- [Google Doc draft](https://docs.google.com/document/d/1mZWkazhVAQn7LaztKeyW0OB87Scd9aYft2KQxVjGEnE/edit?usp=sharing)
- Links to updates to the Terraform language reference [PR-35668](https://github.com/hashicorp/terraform/pull/35668)
- Associated with API docs [PR-699](https://github.com/hashicorp/terraform-docs-common/pull/699) 

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. HCP Terraform ).
- [ ] Description links to related pull requests or issues, if any.

#### Content
- [x] Redirects have been added to `website/redirects.js` for moved, renamed, or deleted pages.
- [ ] API documentation and the API Changelog have been updated. 
- [x] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [x] Pages with related content are updated and link to this content when appropriate.
- [x] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [x] New pages have metadata (page name and description) at the top.
- [ ] New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [ ] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [x] UI elements (button names, page names, etc.) are bolded.
- [x] The Vercel website preview successfully deployed.

#### Reviews
- [ ] I or someone else reviewed the content for technical accuracy.
- [ ] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

[IPE-867]: https://hashicorp.atlassian.net/browse/IPE-867?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: website/docs/cloud-docs/projects/manage.mdx
Comment (brandonc) [inline website/docs/cloud-docs/projects/manage.mdx:109]: I think this stipulation only applies to workspaces -- which is the only resource where you can both inherit and define tags.

File: website/docs/cloud-docs/projects/manage.mdx
Comment (brandonc) [inline website/docs/cloud-docs/projects/manage.mdx:115]:
```suggestion
- Tags can include letters, numbers, colons, hyphens, and underscores.
```

File: website/docs/cloud-docs/projects/manage.mdx
Comment (brandonc) [inline website/docs/cloud-docs/projects/manage.mdx:113]: We should somehow mention the tag count limitations-- 10 per workspace, 10 per project, for a total of 20 per workspace...

File: website/docs/cloud-docs/projects/manage.mdx
Comment (brandonc) [inline website/docs/cloud-docs/projects/manage.mdx:119]: The list is actually `hcp:` `hc:` `ibm:`

File: website/docs/cloud-docs/workspaces/create.mdx
Comment (brandonc) [inline website/docs/cloud-docs/workspaces/create.mdx:98]: Maybe say more because I'm not sure how tags help track resource consumption.

**Review by brandonc — COMMENTED** [review]
A few small things I was able to catch. I think it's great overall. I feel like we should launch with some kind of migration guide that explains why we have two types of tags and when to use each.

Let's hold off on merging during early beta and before we know which TFE the feature will land in

File: website/docs/cloud-docs/projects/manage.mdx
Comment (brandonc) [reply website/docs/cloud-docs/projects/manage.mdx:119]: These show up as links for some reason

Comment (trujillo-adam) [reply website/docs/cloud-docs/projects/manage.mdx:119]: I wasn't sure if we should keep the `ibm` prefix, but I'll add it back in.
Yep, the auto-linking thing is part of our web platform. Nothing to do about it

File: website/docs/cloud-docs/projects/manage.mdx
Comment (trujillo-adam) [inline website/docs/cloud-docs/projects/manage.mdx:109]:
```suggestion
```

File: website/docs/cloud-docs/projects/manage.mdx
Comment (trujillo-adam) [inline website/docs/cloud-docs/projects/manage.mdx:119]:
```suggestion
- You can create up to 10 unique tags per workspace and 10 unique tags per project. As a result, each workspace can have up to 20 tags.
- You cannot use the following strings at the beginning of a tag key:
    - `hcp`
    - `hc`
    - `ibm`
```

File: website/docs/cloud-docs/workspaces/create.mdx
Comment (trujillo-adam) [inline website/docs/cloud-docs/workspaces/create.mdx:98]:
```suggestion
You can define key-value pairs to help you organize your workspaces and use them to sort by cost center, for example, to help you track resource consumption. You can define tags while [creating the workspace](#create-a-workspace) or create and attach them to an existing workspace.
```

**Comment by trujillo-adam** [comment]
>  I feel like we should launch with some kind of migration guide that explains why we have two types of tags and when to use each.

Thanks for the review and the feedback! When you follow the content through the expected user path, especially when you include the API reference and TF core documentation updates, I think it will be clear that there are two types of tags and their functions in the app. My understanding is that when we release this, you really won't be able to create single-value tags anymore, so I not sure what  the content of a migration guide would contain.

File: website/docs/cloud-docs/projects/manage.mdx
Comment (rkoron007) [inline website/docs/cloud-docs/projects/manage.mdx:16]: When mentioning specific permissions I feel like we usually do quotes? I don't love it, but bold doesn't feel right _either_ because we aren't talking about something in UI element here.
```suggestion
- You must be a member of a team with the "Manage all Projects" permission to create a project. Refer to [Organization Permissions](/terraform/cloud-docs/users-teams-organizations/permissions#organization-permissions) for additional information.
```

File: website/docs/cloud-docs/projects/manage.mdx
Comment (rkoron007) [inline website/docs/cloud-docs/projects/manage.mdx:18]: This permission doesn't seem to affect projects at all, just teams? Maybe we can remove it?

File: website/docs/cloud-docs/projects/manage.mdx
Comment (rkoron007) [inline website/docs/cloud-docs/projects/manage.mdx:20]: Again, bold here doesn't feel right since this isn't a UI element on this page? Also how would you feel about a link since the permission is kind of hidden?
```suggestion
To delete tags on a project, you must be member of a team with ["Project Admin" permissions](/terraform/cloud-docs/users-teams-organizations/permissions#project-admin) for that project.
```

File: website/docs/cloud-docs/projects/manage.mdx
Comment (rkoron007) [inline website/docs/cloud-docs/projects/manage.mdx:22]: Same as above
```suggestion
To create tags for a project, you must be member of a team with ["Project Write" permissions](/terraform/cloud-docs/users-teams-organizations/permissions#write-1) for that project.
```

File: website/docs/cloud-docs/projects/manage.mdx
Comment (rkoron007) [inline website/docs/cloud-docs/projects/manage.mdx:17]: I'm not 100% if this is the permission being talking about but I *think* it is, so here is my best guess:
```suggestion
- You must be a member of a team with the [team management "manage" permission](/terraform/cloud-docs/users-teams-organizations/permissions#general-project-permissions) to allow non-secret teams to access a project.
```

File: website/docs/cloud-docs/projects/manage.mdx
Comment (rkoron007) [reply website/docs/cloud-docs/projects/manage.mdx:20]: Also, to match formatting should we either make these list items or remove the bullet points above?

File: website/docs/cloud-docs/projects/manage.mdx
Comment (rkoron007) [inline website/docs/cloud-docs/projects/manage.mdx:41]:
```suggestion
1. Open the **Add key value tags** menu to add tags to your project. Tags are optional key-value pairs that you can use to organize projects. Any workspaces you create within the project inherit project tags. Refer to [Define project tags](#define-project-tags) for additional information.
```

File: website/docs/cloud-docs/projects/manage.mdx
Comment (rkoron007) [inline website/docs/cloud-docs/projects/manage.mdx:43]:
```suggestion
1. Click **+add another tag** to attach any additional tags.
```

File: website/docs/cloud-docs/users-teams-organizations/organizations/index.mdx
Comment (rkoron007) [inline website/docs/cloud-docs/users-teams-organizations/organizations/index.mdx:88]:
```suggestion
You can also view single-value tags that workspace managers added directly to their workspaces. Refer to [Tags](#tags) in the organization settings reference for additional information.
```

File: website/docs/cloud-docs/workspaces/browse.mdx
Comment (rkoron007) [inline website/docs/cloud-docs/workspaces/browse.mdx:20]:
```suggestion
You must be a member of a team with workspace "Read" permissions to view Terraform runs associated with that workspace. Refer to the [permissions reference](/terraform/cloud-docs/users-teams-organizations/permissions#general-workspace-permissions) for additional information.
```

File: website/docs/cloud-docs/workspaces/browse.mdx
Comment (rkoron007) [inline website/docs/cloud-docs/workspaces/browse.mdx:47]: It feels weird to only have one of these bullet points have a bold beginner:
```suggestion
- The tag filter shows a list of tags added to all workspaces, limited to the first 1,000 tags alphabetically. Choosing one or more will show only workspaces tagged with all of the chosen tags.
```

File: website/docs/cloud-docs/workspaces/create.mdx
Comment (rkoron007) [inline website/docs/cloud-docs/workspaces/create.mdx:30]:
```suggestion
* "Manage all projects"
```

File: website/docs/cloud-docs/workspaces/create.mdx
Comment (rkoron007) [inline website/docs/cloud-docs/workspaces/create.mdx:31]:
```suggestion
* "Manage all workspaces"
```

File: website/docs/cloud-docs/workspaces/create.mdx
Comment (rkoron007) [inline website/docs/cloud-docs/workspaces/create.mdx:32]:
```suggestion
* "Admin" permission group for a project
```

File: website/docs/cloud-docs/workspaces/index.mdx
Comment (rkoron007) [inline website/docs/cloud-docs/workspaces/index.mdx:54]: Did you double check for this header name around to make sure no other pages link to it?

**Review by rkoron007 — APPROVED** [review]
Looks **fantastic**! Left some small review notes for style and happy to discuss anything!

File: website/docs/cloud-docs/projects/manage.mdx
Comment (im2nguyen) [inline website/docs/cloud-docs/projects/manage.mdx:74]: i think `!>` corresponds with Warning

```suggestion
!> **Warning:** Automatic destroy plans _do not_ prompt you for apply approval in the HCP Terraform user interface. We recommend only using this setting for development environments.
```

File: website/docs/cloud-docs/projects/manage.mdx
Comment (im2nguyen) [inline website/docs/cloud-docs/projects/manage.mdx:120]: is this an actual technical restriction? if not, we should remove it

```suggestion
```

File: website/docs/cloud-docs/workspaces/create.mdx
Comment (im2nguyen) [inline website/docs/cloud-docs/workspaces/create.mdx:32]: this is what it looks like right now

```suggestion
- **Manage all projects**
- **Manage all workspaces**
- **Admin** permission group for a project.
```

image

File: website/docs/cloud-docs/workspaces/create.mdx
Comment (im2nguyen) [inline website/docs/cloud-docs/workspaces/create.mdx:112]: since this doesn't require numbered list, this will make it easier to parse. also typo (tabs -> tags)

```suggestion
   - Tags must be one or more characters.
   - Tags have a 255 character limit.
   - Tags can include letters, numbers, colons, hyphens, and underscores.
   - Tag values are optional.
```
image

File: website/docs/cloud-docs/workspaces/create.mdx
Comment (im2nguyen) [inline website/docs/cloud-docs/workspaces/create.mdx:122]:
```suggestion
### Configure Terraform variables for VCS workflows
```

File: website/docs/cloud-docs/projects/manage.mdx
Comment (trujillo-adam) [reply website/docs/cloud-docs/projects/manage.mdx:16]: This is referring to a UI element. There is a radio button in the interface with this label.

File: website/docs/cloud-docs/projects/manage.mdx
Comment (trujillo-adam) [reply website/docs/cloud-docs/projects/manage.mdx:18]: It's possible that this has changed and the required permissions were never updated: https://developer.hashicorp.com/terraform/cloud-docs/projects/managing#permissions

@brandonc - is this still accurate?

File: website/docs/cloud-docs/projects/manage.mdx
Comment (trujillo-adam) [reply website/docs/cloud-docs/projects/manage.mdx:17]:
```suggestion
- You must be a member of a team with the **Visible** option enabled under **Visibility** in the organization settings to configure a new team's access to the project. Refer to [Team Visibility](/terraform/cloud-docs/users-teams-organizations/teams#team-visibility) for additional information.
```
Yeah, that wasn't really clear.

File: website/docs/cloud-docs/projects/manage.mdx
Comment (trujillo-adam) [reply website/docs/cloud-docs/projects/manage.mdx:120]: It is. It's in the RFC

File: website/docs/cloud-docs/workspaces/create.mdx
Comment (trujillo-adam) [reply website/docs/cloud-docs/workspaces/create.mdx:114]: Thanks. I need to switch back to VS Code, this JetBrains IDE does all kinds of weird auto-formatting that I don't immediately see.

File: website/docs/cloud-docs/projects/manage.mdx
Comment (impurna) [inline website/docs/cloud-docs/projects/manage.mdx:42]: "private beta" is the approved release stage name.

Comment (trujillo-adam) [reply website/docs/cloud-docs/projects/manage.mdx:42]:
```suggestion
    ~> **Adding project tags is in private beta and unavailable for some users.** Skip to step 8 if your interface does not include elements for adding tags. Contact your HashiCorp representative for information about participating in the limited beta.  
```

File: website/docs/cloud-docs/projects/manage.mdx
Comment (trujillo-adam) [inline website/docs/cloud-docs/projects/manage.mdx:42]:
```suggestion
    ~> **Adding project tags is in private beta and unavailable for some users.** Skip to step 8 if your interface does not include elements for adding tags. Contact your HashiCorp representative for information about participating in the private beta.  
```

File: website/docs/cloud-docs/projects/manage.mdx
Comment (trujillo-adam) [inline website/docs/cloud-docs/projects/manage.mdx:107]:
```suggestion
~> **Adding project tags is in private beta and unavailable for some users.** Contact your HashiCorp representative for information about participating in the private beta.  
```

File: website/docs/cloud-docs/users-teams-organizations/organizations/index.mdx
Comment (trujillo-adam) [inline website/docs/cloud-docs/users-teams-organizations/organizations/index.mdx:74]:
```suggestion
~> **Reserved tags is in private beta and unavailable for some users.** Contact your HashiCorp representative for information about participating in the private beta.
```

File: website/docs/cloud-docs/workspaces/tags.mdx
Comment (trujillo-adam) [inline website/docs/cloud-docs/workspaces/tags.mdx:16]:
```suggestion
~> **Adding tags stored as key-value pairs is in private beta and unavailable for some users.** Contact your HashiCorp representative for information about participating in the private beta.
```

File: website/docs/cloud-docs/workspaces/tags.mdx
Comment (trujillo-adam) [inline website/docs/cloud-docs/workspaces/tags.mdx:27]:
```suggestion
~> **Reserved tags and project tags are in private beta and unavailable for some users.** Contact your HashiCorp representative for information about participating in the private beta.
```

**Pull request merged**
